spec.bs

```diff
@@ -678,15 +684,23 @@ The [=steps to respond to a payment request=] for this payment method, for a giv
 
     1. [=list/Append=] |descriptor| to |publicKeyOpts|["{{PublicKeyCredentialRequestOptions/allowCredentials}}"].
 
-1. Let |outputCredential| be the result of running the algorithm to
+1. Let |responseDetails| be the result of running the algorithm to
     [=Request a Credential=], passing «["{{CredentialRequestOptions/publicKey}}"
     → |publicKeyOpts|]».
 
     Note: Chrome's initial implementation does not pass the full `data.credentialIds` list to [=Request a Credential=]. Instead, it chooses one credential in the list that matches the current device and passes only that in.
 
     Note: This triggers [[webauthn-3]]'s [[webauthn-3#sctn-getAssertion|Get]] behavior
 
-1. Return |outputCredential|.
+1. Insert a {{boolean}} attribute named `iconShown` to |responseDetails| with
```

Comment:
This is not how we should do this :).

This information should be communicated via the [CollectedClientAdditionalPaymentData dictionary](https://w3c.github.io/secure-payment-confirmation/#sctn-collectedclientadditionalpaymentdata-dictionary), which is created during [client extension processing](https://w3c.github.io/secure-payment-confirmation/#client-extension-processing-authentication).

I can think of a few ways to do this, from simplest to very complex:

1. Actually don't give RPs any sort of indication as to whether or not the icon was shown. Instead, setting `iconMustBeShown` to `false` (which they can verify) is an indication that they don't get to know whether or not it was shown.

1. In the [steps to check if a payment can be made](https://w3c.github.io/secure-payment-confirmation/#sctn-steps-to-check-if-a-payment-can-be-made), if the fetch/decode fails then set `data["instrument"]["icon"]` to an empty string. RPs can then verify that it is empty and consider that as "icon wasn't shown".

1. The hard way:

    1. Turn `PaymentCredentialInstrument` into three dictionaries: `PaymentCredentialInstrumentInput`, `PaymentCredentialInstrumentOutput`, and a core `PaymentCredentialInstrument` that they extend.
    1. Add `iconMustBeShown` to `PaymentCredentialInstrumentInput`
    1. Add `iconWasShown` to `PaymentCredentialInstrumentOutput`.
    1. Change `SecurePaymentConfirmationRequest` to have a `PaymentCredentialInstrumentInput`.
    1. Change `AuthenticationExtensionsPaymentInput` and `CollectedClientAdditionalPaymentData` to have a `PaymentCredentialInstrumentOutput`.
    1. In the [steps to respond to a payment request](https://w3c.github.io/secure-payment-confirmation/#sctn-steps-to-respond-to-a-payment-request), step 2, set `instrument` to be a `PaymentCredentialInstrumentOutput` whose `iconWasShown` field is set according to whether or not the icon was shown or not.

Currently, I'd suggest we do approach 2, and we can talk through the alternatives in general review :)